[linux-6.6.y] update ITMT support patch for Zhaoxin CPUs

leoliu-oc opened the pull request:
Include 3 patches:

0001-Revert-cpufreq-ACPI-add-ITMT-support-when-CPPC-enabl.patch
0002-cpufreq-ACPI-add-ITMT-support-when-CPPC-enabled.patch
0003-cpufreq-acpi-cpufreq-Zhaoxin-fix-incorrect-max-freq-.patch

Roll back the previous old patch, update to the new patch version,
and fix the issue of mismatched max freq.

opsiff reviewed (approved):
/lgtm

和 https://lore.kernel.org/all/20230808111325.8600-1-TonyWWang-oc@zhaoxin.com/ 基本一致

leoliu-oc committed:
Revert "cpufreq: ACPI: add ITMT support when CPPC enabled"

This reverts commit e28f9776e8691d40eb210b15f100241620b95f45.

Signed-off-by: LeoLiu-oc <leoliu-oc@zhaoxin.com>

leoliu-oc committed:
cpufreq: ACPI: add ITMT support when CPPC enabled

The _CPC method can get per-core highest frequency.
The highest frequency may varies between cores which mean cores can
running at different max frequency, so can use it as a core priority
and give a hint to scheduler in order to put critical task to the
higher priority core.

Signed-off-by: LeoLiu-oc <leoliu-oc@zhaoxin.com>

leoliu-oc committed:
cpufreq: acpi-cpufreq: Zhaoxin: fix incorrect max-freq issue

Workaround for incorrect max frequency calculation on Zhaoxin/Centaur CPUs.

Signed-off-by: LeoLiu-oc <leoliu-oc@zhaoxin.com>

opsiff commented:
/lgtm
